### PR TITLE
User Roles Implementation

### PR DESCRIPTION
**Pull request by deye9**
User Roles Implementation

### CODE REVIEW

File: libraries/src/main/java/com/foodie/kafka/service/KafKaTopicListeners.java [hunk 1]
```diff
@@ -15,11 +15,11 @@ public void listenUserTopic(Object message) {
 
     @KafkaListener(topics = "foodie-users", groupId = "foodie-users-group")
     public void listenUsersTopic(Object message) {
-    log.info("Received message in foodie-users: " + message);
+        log.info("Received message in foodie-users: " + message);
```
Comment (sfagade) [inline libraries/src/main/java/com/foodie/kafka/service/KafKaTopicListeners.java:18]: Please fix, this form of string concatination actually uses more resources

File: services/user-management/src/main/java/com/foodie/user/contracts/RolePermitRequest.java [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
 
+import com.foodie.user.model.Permission;
+
 import jakarta.validation.constraints.NotEmpty;
 import jakarta.validation.constraints.NotNull;
 
 public record RolePermitRequest(
         @NotEmpty String id,
-        @NotNull PermissionRequest permission) {
+        @NotNull Permission permission) {
```
Comment (sfagade) [inline services/user-management/src/main/java/com/foodie/user/contracts/RolePermitRequest.java:10]: What you had before is better

File: services/user-management/src/main/java/com/foodie/user/controllers/RolePermissionController.java [hunk 3]
```diff
@@ -19,16 +18,19 @@
...
 
 @Slf4j
 @RestController
 @CrossOrigin
 @RequestMapping("/api/v1/security/{role}/permissions")
 public class RolePermissionController {
 
-    private final RoleService roleService;
-    private final PermissionService permissionService;
+    public final RoleService roleService;
```
Comment (sfagade) [inline services/user-management/src/main/java/com/foodie/user/controllers/RolePermissionController.java:32]: Why are you making these `public`? They should be `prive`

File: services/user-management/src/main/java/com/foodie/user/repositories/UserRoleRepository.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+@Repository
+public interface UserRoleRepository extends FoodieBaseRepository<UserRole, UUID> {
+
+    @Modifying
+    @Transactional
+    @Query("update UserRole ur set ur.deletedAt = CURRENT_TIMESTAMP where ur.userId = :userId")
+    void deleteAllByUserId(@Param("userId") UUID id);
+
+    @Query("SELECT ur FROM UserRole ur JOIN ur.userId u WHERE u.id = :userId AND ur.deletedAt IS NULL")
```
Comment (sfagade) [inline services/user-management/src/main/java/com/foodie/user/repositories/UserRoleRepository.java:27]: If your entity classes are created correctly then you wouldn't need to have this queries here. I imagine you would be able to accomplish the same goals using magic methods

**Review by sfagade — APPROVED** [review]

File: libraries/src/main/java/com/foodie/kafka/service/KafKaTopicListeners.java (hunk 1 above)
Comment (deye9) [reply libraries/src/main/java/com/foodie/kafka/service/KafKaTopicListeners.java:18]: Done

File: services/user-management/src/main/java/com/foodie/user/contracts/RolePermitRequest.java (hunk 2 above)
Comment (deye9) [reply services/user-management/src/main/java/com/foodie/user/contracts/RolePermitRequest.java:10]: Objects.NotNull is better suited for objects from what I read

File: services/user-management/src/main/java/com/foodie/user/controllers/RolePermissionController.java (hunk 3 above)
Comment (deye9) [reply services/user-management/src/main/java/com/foodie/user/controllers/RolePermissionController.java:32]: That was an oversight

File: services/user-management/src/main/java/com/foodie/user/repositories/UserRoleRepository.java (hunk 4 above)
Comment (deye9) [reply services/user-management/src/main/java/com/foodie/user/repositories/UserRoleRepository.java:27]: Due to the onetomany mapping. This query is needed

**Pull request merged**